### DISCUSSION TITLE
Rendering filter form manually adds "Value" label

### DISCUSSION THREAD

**Discussion by Pyrol** [Kreyu/data-table-bundle#124] [Q&A]
Hi, I found strange behaviour while rendering `filtration_form` manually. 
When I render row with other label `form_row(field, {'attr' : {'label:'test-label'}})` I can see the change but I still can see "Value" below, the same behaviour is when I render only  `form_widget(field)`. Do you know what may be the cause?

The same behaviour is without BS theme. 

I use it because I get table from other path, submit button works on current path and I'm unable/don't know how to set action parameter when I render it with `data_table_filters_form(form, variables)` .

```TWIG
    {% form_theme dt.vars.filtration_form.vars.form 'bootstrap_5_layout.html.twig' %}
    {{form_start(dt.vars.filtration_form.vars.form)}}
    <div class="row">
        {% for field in dt.vars.filtration_form.vars.form %}
        <div class="col-3">
            {{ dump(field) }}
            {{ form_label(field , "Custom Label") }}
            {{ form_row(field) }}  {# I can  change field name label but "Value" still appears below#}
            {{ form_widget(field) }} {# With widget I also get "Value" label #}
        </div>
        {% endfor %}
    </div>
    {{ form_end(dt.vars.filtration_form.vars.form) }}
```

![image](https://github.com/user-attachments/assets/855c6ad9-5d44-4061-b07d-4a8e03c73fb1)

**Comment by Pyrol** [comment]
(If you could explain it more technically would be great)
You need to reach for actual children which is  ```dt.vars.filtration_form.vars.form.children``` (in topic post I iterated wrong) and then call ```{{ form_widget(field.value) }}``` - I thought that simply calling form() function will do because of Symfonys form form_name[field]=val but there is additional [value] in the name of field `form[field][value]=val

**Comment by Kreyu** [comment]
Hey @Pyrol!

> I thought that simply calling form() function will do because of Symfonys form form_name[field]=val but there is additional [value] in the name of field `form[field][value]=val

I am unsure why calling `form` wouldn't work, but let me explain the `value` part. That's because each filter _can_ consist of two fields - one for the value, and one for the operator, which is not rendered by default. The operator is used for more advanced filtering. For example, user can select "not contains" in the string filter instead of the default "contains". With operator selector visible, the URL parameter would be `form[field][value]=val&form[field][operator]=not-contains`. 

So, if you have two filters - `firstName` and `lastName`, the form will look like this:

- form
    - firstName
        - value
        - operator
    -  lastName
        - value
        - operator

I hope that explains the filtration form structure. More about operators:
- [Displaying operator selector](https://data-table-bundle.swroblewski.pl/docs/components/filters.html#displaying-operator-selector)
- [Changing default operator](https://data-table-bundle.swroblewski.pl/docs/components/filters.html#changing-default-operator)
- [Operator form type and options](https://data-table-bundle.swroblewski.pl/docs/components/filters.html#operator-form-type-and-options)
